api/acme/directory.go

```diff
@@ -0,0 +1,155 @@
+package acme
+
+import "fmt"
+
+// Directory represents an ACME directory for configuring clients.
+type Directory struct {
+	NewNonce   string
+	NewAccount string
+	NewOrder   string
+	NewAuthz   string
+	RevokeCert string
+	KeyChange  string
+	Directory  string
+	prefix     string
+	dns        string
+	ops        *DirectoryOptions
+}
+
+// DirectoryOptions options with which to create a new Directory type.
+type DirectoryOptions struct {
+	Nonce       string
+	NewNonce    string
+	Account     string
+	NewAccount  string
+	Order       string
+	NewOrder    string
+	Authz       string
+	NewAuthz    string
+	Challenge   string
+	RevokeCert  string
+	KeyChange   string
+	Certificate string
+	Directory   string
+}
+
+var defaultDirOptions = DirectoryOptions{
+	Nonce:       "nonce",
+	NewNonce:    "new-nonce",
+	Account:     "account",
+	NewAccount:  "new-account",
+	Order:       "order",
+	NewOrder:    "new-order",
+	Authz:       "authz",
+	NewAuthz:    "new-authz",
+	Challenge:   "challenge",
+	RevokeCert:  "revoke-cert",
+	KeyChange:   "key-change",
+	Certificate: "cert",
+	Directory:   "directory",
+}
+
+// NewDirectory returns a new Direcotyr type.
+func NewDirectory(dns, prefix string, ops *DirectoryOptions) *Directory {
+	if ops == nil {
+		ops = &defaultDirOptions
+	}
+
+	return &Directory{
+		prefix:     prefix,
+		dns:        dns,
+		ops:        ops,
+		NewNonce:   fmt.Sprintf("/%s/%s", prefix, ops.NewNonce),
+		NewAccount: fmt.Sprintf("/%s/%s", prefix, ops.NewAccount),
+		NewOrder:   fmt.Sprintf("/%s/%s", prefix, ops.NewOrder),
+		NewAuthz:   fmt.Sprintf("/%s/%s", prefix, ops.NewAuthz),
+		RevokeCert: fmt.Sprintf("/%s/%s", prefix, ops.RevokeCert),
+		KeyChange:  fmt.Sprintf("/%s/%s", prefix, ops.KeyChange),
+		Directory:  fmt.Sprintf("/%s/%s", prefix, ops.Directory),
+	}
+}
+
+// GetAccount returns the location for retrieving Account info by ID.
+func (d *Directory) GetAccount(id string, abs bool) string {
+	p := fmt.Sprintf("/%s/%s/%s", d.prefix, d.ops.Account, id)
+	if abs {
+		p = fmt.Sprintf("https://%s%s", d.dns, p)
+	}
+	return p
```

Comment:
It looks like this code is repeated, we can unify it using a new method with the necessary arguments.